projects/ion/src/lib/avatar/avatar.component.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { render, screen } from '@testing-library/angular';
+
+import { IonAvatarComponent } from './avatar.component';
+import { AvatarSize, AvatarType } from './types';
+import { DefaultImageDirective } from './defaultImage.directive';
+import { IonIconComponent } from '../icon';
+
+async function sut(
+  props: Partial<IonAvatarComponent> = {}
+): Promise<HTMLElement> {
+  await render(IonAvatarComponent, {
+    componentProperties: props,
+    providers: [DefaultImageDirective],
+    imports: [IonIconComponent],
+  });
+  return screen.getByTestId('ion-avatar');
+}
+
+describe('Avatar', () => {
+  describe('Basics', () => {
+    const sizes: AvatarSize[] = ['lg', 'md', 'sm', 'xs'];
+    it.each(sizes)('should have size-%s class', async (size: AvatarSize) => {
```

Comment:
Here you can verify if the element has the data-type attribute with its respective value, using component fixture.nativeElement